<?xml version="1.0"?>
<launch>

        <node pkg="tf2_ros" type="static_transform_publisher" name="map_joint" args="odomPose_x  odomPose_y  odomPose_z  odomPose_roll  odomPose_pitch  odomPose_yaw   map   odom"/>

        <node pkg="tf" type="static_transform_publisher" name="base_link_to_sim_base_link" args="0 0 0 0 0 0 base_link /puzzlebot_1/base_link 100"/>
        <node pkg="tf" type="static_transform_publisher" name="base_link_to_sim_lidar_link" args="0 0 0 0 0 0 /puzzlebot_1/base_link /puzzlebot_1/lidar_link 100"/>
        <!-- <node pkg="tf" type="static_transform_publisher" name="simbase_to_base_footprint" args="0 0 0 0 0 0 /puzzlebot_1/base_link /puzzlebot_1/base_footprint 100"/> -->
        <node pkg="tf" type="static_transform_publisher" name="odom_to_sim" args="0 0 0 0 0 0 odom world 100"/>

        <arg name="puzzlebot_sim_model" default="$(find localisation)/urdf/puzzlebot_jetson_ed_urdf.urdf"/>
        <param name="robot_description" command="cat $(arg puzzlebot_sim_model)" />

        <!-- <node pkg="localisation" type="puzzlebot_sim.py" name="simulator" output="screen"/> -->
        <!-- <node pkg="localisation" type="localisation.py" name="localisation_node"/> -->
        <node pkg="localisation" type="joint_state_pub.py" name="joint_publisher_node"/>
        <!-- <node pkg="localisation" type="controller_sim_bug2.py" name="controller_node" output="screen"/> -->
        

        <node pkg="robot_state_publisher" type="robot_state_publisher" name="simulated_state_publisher" output="screen"/>
        <!-- <node name="rviz" pkg="rviz" type="rviz"/> -->
        <arg name="rvizconfig" default="$(find localisation)/rviz/localization.rviz" />
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" /> 
        <include file="$(find puzzlebot_gazebo)/launch/puzzlebot_gazebo.launch" />
</launch>
